src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -295,9 +296,18 @@ public IList<string> Merge (List<TypeDefinition> subclasses, List<string> select
 						if (fromCode == null)
 							continue;
 
-						if (!t.Methods.Where (m => m.IsConstructor).Cast<MethodDefinition> ().Any (c => !c.HasParameters && c.IsPublic))
-							throw new InvalidOperationException (string.Format ("The type '{0}' needs to have a public default constructor.",
-										t.FullName));
+						IEnumerable <MethodDefinition> constructors = t.Methods.Where (m => m.IsConstructor).Cast<MethodDefinition> ();
+						if (!constructors.Any (c => !c.HasParameters && c.IsPublic)) {
+							string location = null;
+							SequencePoint sourceLocation = FindSource (constructors);
+
+							if (sourceLocation != null && sourceLocation.Document?.Url != null)
+								location = $" at {sourceLocation.Document.Url}:{sourceLocation.StartLine}";
+							else
+								location = ".";
+							log.LogCodedError ("XA4213", $"The type '{t.FullName}' needs to have a public default constructor{location}");
```

Comment:
In this case, we should instead use [`TaskLoggingHelper.LogError()`](https://msdn.microsoft.com/en-us/library/ms126325(v=vs.121).aspx), and provide all of the arguments:

```csharp
if (sourceLocation != null && sourceLocation.Document?.Url != null) {
    log.LogError (
            subcategory:      string.Empty,
            errorCode:        "XA4213",
            helpKeyword:      string.Empty,
            file:             sourceLocation.Document.Url
            lineNumber:       sourceLocation.StartLine,
            columnNumber:     sourceLocation.StartColumn,
            endLineNumber:    sourceLocation.EndLine,
            endColumnNumber:  sourceLocation.EndColumn,
            message:          $"The type '{t.FullName}' must provide a public default constructor.")
} else {
    log.LogCodedError ("XA4213", $"The type '{t.FullName}' needs to have a public default constructor");
}
```